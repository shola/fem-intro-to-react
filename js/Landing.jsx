// @flow
import React from 'react';
import MyTitle from './MyTitle';

function Landing(): React.Element<any> {
    return (
        <div className="app">
            <div className="landing">
                <MyTitle title="Welcome to my App!" />
                <input className="search" type="text" placeholder="search" />
                <button className="browse-all"> or Browse All</button>
            </div>
        </div>
    );
}

export default Landing;
