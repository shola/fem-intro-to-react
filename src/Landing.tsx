import * as React from 'react';
import { Link } from 'react-router';
import MyTitle from './MyTitle';

/**
 * Render landing page
 */
const Landing = () => (
    <div className="home-info">
        <MyTitle title="Welcome to my App!" />
        <input className="search" type="text" placeholder="search" />
        <Link to="/search" className="browse-all"> or Browse All</Link>
    </div>
);

export default Landing;
