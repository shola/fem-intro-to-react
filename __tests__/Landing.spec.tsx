import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as Renderer from 'react-test-renderer';
import Landing from '../src/Landing';

describe('Landing Page', () => {
    it('Check snapshot for unexpected changes', () => {
        const component = Renderer.create(<Landing />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
