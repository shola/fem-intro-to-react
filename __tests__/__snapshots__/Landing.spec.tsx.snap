// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Check snapshot for unexpected changes 1`] = `
<div
  className="home-info"
>
  <h1
    className="title"
    style={
      Object {
        "color": "red",
      }
    }
  >
    Welcome to my App!
  </h1>
  <input
    className="search"
    placeholder="search"
    type="text"
  />
  <a
    className="browse-all"
    onClick={[Function]}
    style={Object {}}
  >
     or Browse All
  </a>
</div>
`;
